pkgs/development/tools/build-managers/bazel/default.nix

```diff
@@ -13,60 +12,54 @@ let
     platforms = platforms.linux;
   };
 
-  bootstrapEnv = buildFHSUserEnv {
-    name = "bazel-bootstrap-env";
+  name = "bazel-${version}";
 
-    targetPkgs = pkgs: [ ];
-
-    inherit meta;
-  };
-
-  bazelBinary = stdenv.mkDerivation rec {
-
-    name = "bazel-${version}";
-
-    src = fetchurl {
-      url = "https://github.com/bazelbuild/bazel/releases/download/${version}/bazel-${version}-dist.zip";
-      sha256 = "1fwfahkqi680zyxmdriqj603lpacyh6cg6ff25bn9bkilbfj2anm";
-    };
-
-    sourceRoot = ".";
-
-    patches = [ ./java_stub_template.patch ];
-
-    packagesNotFromEnv = [
-        stdenv.cc stdenv.cc.cc.lib jdk which zip unzip binutils ];
-    buildInputs = packagesNotFromEnv ++ [ bootstrapEnv makeWrapper ];
-
-    buildTimeBinPath = stdenv.lib.makeBinPath packagesNotFromEnv;
-    buildTimeLibPath = stdenv.lib.makeLibraryPath packagesNotFromEnv;
-
-    runTimeBinPath = stdenv.lib.makeBinPath [ jdk stdenv.cc.cc ];
-    runTimeLibPath = stdenv.lib.makeLibraryPath [ stdenv.cc.cc.lib ];
-
-    buildWrapper = writeScript "build-wrapper.sh" ''
-      #! ${stdenv.shell} -e
-      export PATH="${buildTimeBinPath}:$PATH"
-      export LD_LIBRARY_PATH="${buildTimeLibPath}:$LD_LIBRARY_PATH"
-      ./compile.sh
-    '';
-
-    buildPhase = ''
-      bazel-bootstrap-env ${buildWrapper}
-    '';
-
-    installPhase = ''
-      mkdir -p $out/bin
-      cp output/bazel $out/bin/
-      wrapProgram $out/bin/bazel \
-          --suffix PATH ":" "${runTimeBinPath}" \
-          --suffix LD_LIBRARY_PATH ":" "${runTimeLibPath}"
-    '';
-
-    dontStrip = true;
-    dontPatchELF = true;
-
-    inherit meta;
+  src = fetchurl {
+    url = "https://github.com/bazelbuild/bazel/releases/download/${version}/bazel-${version}-dist.zip";
+    sha256 = "1fwfahkqi680zyxmdriqj603lpacyh6cg6ff25bn9bkilbfj2anm";
   };
 
-in bazelBinary
+  sourceRoot = ".";
+
+  patches = [ ./bin_to_env.patch ];
```

Comment:
The patch is quite big. Would it make sense to manage this patch external in a git like we do [for](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/interpreters/ruby/default.nix#L14) [ruby](https://github.com/zimbatm/rubygems/compare/v2.6.6...v2.6.6-nix.patch) to make rebasing easier?